parser/parser/cc/lexer.rl

```diff
@@ -558,7 +558,12 @@ static bool eof_codepoint(char c) {
 token_t lexer::advance_() {
   if (!token_queue.empty()) {
     token_t token = token_queue.front();
-    token_queue.pop_front();
+    if (token->type() == token_type::eof) {
+      // spin on EOF forever, even if advance is called repeatedly
```

Comment:
This used to be the previous behavior as well, by nature of how the `p` and `pe` variables would check.

There was no limit to the number of times that you could call `advance()` (and thus `advance_()`). If `p == pe`, ragel would just exit the `%exec` block of generated code, skipping to the bottom of the `advance_()` function, which always exited by just returning an `eof` token.

I only wanted to emit the `tBEFORE_EOF` function **once**, not do something like accidentally alternate between `..., tBEFORE_EOF, eof, tBEFORE_EOF, eof`. Doing this trick with the `token_queue` seemed like the cleanest trick.